src/main/java/com/onsdigital/performance/reporter/google/PercentageComparison.java

```diff
@@ -0,0 +1,161 @@
+package com.onsdigital.performance.reporter.google;
+
+import com.onsdigital.performance.reporter.interfaces.CompositeMetricProvider;
+import com.onsdigital.performance.reporter.interfaces.MetricProvider;
+import com.onsdigital.performance.reporter.model.Metric;
+import com.onsdigital.performance.reporter.model.MetricDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Enables a percentage of two google analytics dimensions to be calculated.
+ * <p>
+ * A single MetricDefinition is provided which defines a base analytics query.
+ * Two separate dimension values are provided in the meta field of the Metric.
+ * <p>
+ * A single metric is returned which provides a percentage of the first dimension compared to the other.
+ * <p>
+ * Example: % of users that have downloaded from a dataset page.
+ * Query: User sessions
+ * Dimension A: User sessions that have looked at a dataset page and have a download event
+ * Dimension B: User sessions that have looked at a dataset page
+ */
+public class PercentageComparison implements CompositeMetricProvider {
+
+    private static final String ComparisonFieldKey = "comparisonField";
+    private MetricProvider metricProvider;
+
+    @Override
+    public Metric getMetric(MetricDefinition metricDefinition) throws IOException {
+
+        Metric metricA = getInputMetric(metricDefinition, "comparisonValueA");
+        Metric metricB = getInputMetric(metricDefinition, "comparisonValueB");
+
+        return createOutputMetric(metricA, metricB);
+    }
+
+    private Metric getInputMetric(MetricDefinition metricDefinition, String dimension) throws IOException {
+        // Take the given metric definition and copy the query from it into two separate definitions.
+        MetricDefinition request = createDefinitionUsingFilter(metricDefinition, dimension);
+
+        // Call out to Google analytics for the individual metric data to compare.
+        return metricProvider.getMetric(request);
+    }
+
+    private static Metric createOutputMetric(Metric metricA, Metric metricB) {
+        // Create the result metric
+        Metric metric = new Metric();
+        metric.values = new ArrayList<>();
+
+        if (metricA != null && metricA.columns != null)
+            metric.columns = metricA.columns; // copy column headers
+
+        metric.values = calculateMetricValues(metricA, metricB);
+
+        return metric;
+    }
+
+    private static List<List<String>> calculateMetricValues(Metric metricA, Metric metricB) {
+        List<List<String>> values = new ArrayList<>();
+
+        if (metricA != null && metricB != null) {
+            for (int rowIndex = 0; rowIndex < metricA.values.size(); rowIndex++) {
+                List<String> rowA = metricA.values.get(rowIndex);
```

Comment:
Good point. I have changed it to throw an exception if there are any differences in length of columns or rows. Any differences would imply an issue with the query input.